src/main/java/fact/coordinates/HorizontalCoordinate.java

```diff
@@ -0,0 +1,118 @@
+package fact.coordinates;
+
+import org.apache.commons.math3.geometry.euclidean.threed.Rotation;
+import org.apache.commons.math3.geometry.euclidean.threed.Vector3D;
+
+import java.time.ZonedDateTime;
+
+/**
+ * Created by maxnoe on 22.05.17.
+ */
+public class HorizontalCoordinate implements CelestialCoordinate {
+
+    private final double zenithRad;
+    private final double azimuthRad;
+
+    public HorizontalCoordinate(double zenithRad, double azimuthRad) {
+        this.zenithRad = zenithRad;
+        this.azimuthRad = azimuthRad;
+    }
+
+    public static HorizontalCoordinate fromDegrees(double zenithDeg, double azimuthDeg) {
+        return new HorizontalCoordinate(Math.toRadians(zenithDeg), Math.toRadians(azimuthDeg));
+    }
+
+    public double greatCircleDistance(HorizontalCoordinate other) {
+        return CelestialCoordinate.greatCircleDistance(
+                this.getAltitudeRad(), this.getAzimuthRad(),
+                other.getAltitudeRad(), other.getAzimuthRad()
+        );
+    }
+
+
+    /**
+     * Implementation of the formulas from
+     * https://en.wikipedia.org/wiki/Celestial_coordinate_system#Equatorial_.E2.86.90.E2.86.92_horizontal
+     *
+     * @param observationTime
+     * @param earthLocation
+     * @return HorizontalCoordinate for given time and location;
+     */
+    public EquatorialCoordinate toEquatorial(ZonedDateTime observationTime, EarthLocation earthLocation) {
+
+        double gst = CoordinateUtils.datetimeToGST(observationTime);
+        double azimuthSouth = this.getAzimuthRad() - Math.PI;
+        double alt = this.getAltitudeRad();
+
+        double lat = earthLocation.getLatitudeRad();
+        double lon = earthLocation.getLongitudeRad();
+
+        double hourAngle = Math.atan2(
+                Math.sin(azimuthSouth),
+                Math.cos(azimuthSouth) * Math.sin(lat) + Math.tan(alt) * Math.cos(lat)
+        );
+
+        double declination = Math.asin(Math.sin(lat) * Math.sin(alt) - Math.cos(lat) * Math.cos(alt) * Math.cos(azimuthSouth));
+
+        double ra = gst + lon - hourAngle;
+
+        return new EquatorialCoordinate(ra, declination);
+    }
+
+    /**
+     * Returns position of the source in the camera from the given pointing position
```

Comment:
I find this confusing. Does it return the position of a source in the camera or does it transform the horizontal coordinate called `pointingPosition` to camera coordinates?